jenkins/env_setup/Jenkinsnodedetails.groovy

```diff
@@ -0,0 +1,47 @@
+def count = 0
+try {
+    this.println('Sr.no,Node_name,Host,Label,User,UserId')
+    for (Node n : Jenkins.get().getNodes()) {
+        try {
+            if (n.getComputer().isOffline() == true) {
+                count ++
+                hudson.plugins.jobConfigHistory.ConfigInfo config = n.getComputer().actions[0].getSlaveConfigs()[0]
+                this.println('' + count + ',' + n.name + ',' + n.launcher.host + ',' + n.getLabelString() + ',' + config.getUser() + ',' + config.getUserID())
+            }
+        }
+        catch (Exception e) {
+            this.println('' + count + ' ' + e)
+        }
+    }
+}
+catch (Exception e) {
+    this.println('Inside catch 2. ' + e)
```

Comment:
Please add proper message instead of `Inside catch 2`. I guess it is for your debugging.

Reply:
In some cases, the information is not available in Node apis. It is used for handling these exceptions, Can I keep it empty?